Unable to use map in a typescript project

### Description

After adding map into a vanilla-ts Vite project, the project can no longer be built

### Actual outcome

```
node_modules/@vaadin/map/src/vaadin-map.d.ts:8:31 - error TS2307: Cannot find module '@vaadin/vaadin-themable-mixin' or its corresponding type declarations.

8 import { ThemableMixin } from '@vaadin/vaadin-themable-mixin';
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/ol/geom/Circle.d.ts:55:5 - error TS2416: Property 'getCoordinates' in type 'Circle' is not assignable to the same property in base type 'SimpleGeometry'.
  Type '() => null' is not assignable to type '() => any[]'.
    Type 'null' is not assignable to type 'any[]'.

55     getCoordinates(): null;
       ~~~~~~~~~~~~~~

node_modules/ol/geom/SimpleGeometry.d.ts:27:53 - error TS2694: Namespace '"/Users/artur/test/vite-project/node_modules/ol/geom/GeometryLayout"' has no exported member 'default'.

27     protected layout: import("./GeometryLayout.js").default;
                                                       ~~~~~~~

node_modules/ol/ImageBase.d.ts:32:48 - error TS2694: Namespace '"/Users/artur/test/vite-project/node_modules/ol/ImageState"' has no exported member 'default'.

32     protected state: import("./ImageState.js").default;
                                                  ~~~~~~~

node_modules/ol/Tile.d.ts:138:47 - error TS2694: Namespace '"/Users/artur/test/vite-project/node_modules/ol/TileState"' has no exported member 'default'.

138     protected state: import("./TileState.js").default;
                                                  ~~~~~~~

node_modules/typescript/lib/lib.dom.d.ts:4482:101 - error TS2344: Type 'HTMLElementTagNameMap[K]' does not satisfy the constraint 'Element'.
  Type 'HTMLElement | HTMLImageElement | HTMLVideoElement | HTMLCanvasElement | HTMLAnchorElement | ... 64 more ... | Map' is not assignable to type 'Element'.
    Type 'Map' is missing the following properties from type 'Element': attributes, classList, className, clientHeight, and 160 more.

4482     getElementsByTagName<K extends keyof HTMLElementTagNameMap>(qualifiedName: K): HTMLCollectionOf<HTMLElementTagNameMap[K]>;
                                                                                                         ~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/typescript/lib/lib.dom.d.ts:4770:101 - error TS2344: Type 'HTMLElementTagNameMap[K]' does not satisfy the constraint 'Element'.
  Type 'HTMLElement | HTMLImageElement | HTMLVideoElement | HTMLCanvasElement | HTMLAnchorElement | ... 64 more ... | Map' is not assignable to type 'Element'.

4770     getElementsByTagName<K extends keyof HTMLElementTagNameMap>(qualifiedName: K): HTMLCollectionOf<HTMLElementTagNameMap[K]>;
                                                                                                         ~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/typescript/lib/lib.dom.d.ts:10038:87 - error TS2344: Type 'HTMLElementTagNameMap[K]' does not satisfy the constraint 'Node'.
  Type 'HTMLElement | HTMLImageElement | HTMLVideoElement | HTMLCanvasElement | HTMLAnchorElement | ... 64 more ... | Map' is not assignable to type 'Node'.
    Type 'Map' is missing the following properties from type 'Node': baseURI, childNodes, firstChild, isConnected, and 46 more.

10038     querySelectorAll<K extends keyof HTMLElementTagNameMap>(selectors: K): NodeListOf<HTMLElementTagNameMap[K]>;
                                                                                            ~~~~~~~~~~~~~~~~~~~~~~~~


Found 8 errors.
```


### Minimal reproducible example

```
npm init vite #pick vanilla-ts
cd vite-project
npm i @vaadin/map
echo "import '@vaadin/map'" >> src/main.ts 
npm run build
```

### Environment

- Version of the webcomponent / Vaadin platform.  23 beta 1
- OS. mac os Big sur

The following compilation errors are related to the upstream library:

```
node_modules/ol/geom/Circle.d.ts:55:5 - error TS2416: Property 'getCoordinates' in type 'Circle' is not assignable to the same property in base type 'SimpleGeometry'.
  Type '() => null' is not assignable to type '() => any[]'.
    Type 'null' is not assignable to type 'any[]'.

55     getCoordinates(): null;
       ~~~~~~~~~~~~~~

node_modules/ol/geom/SimpleGeometry.d.ts:27:53 - error TS2694: Namespace '"/Users/artur/test/vite-project/node_modules/ol/geom/GeometryLayout"' has no exported member 'default'.

27     protected layout: import("./GeometryLayout.js").default;
                                                       ~~~~~~~

node_modules/ol/ImageBase.d.ts:32:48 - error TS2694: Namespace '"/Users/artur/test/vite-project/node_modules/ol/ImageState"' has no exported member 'default'.

32     protected state: import("./ImageState.js").default;
                                                  ~~~~~~~

node_modules/ol/Tile.d.ts:138:47 - error TS2694: Namespace '"/Users/artur/test/vite-project/node_modules/ol/TileState"' has no exported member 'default'.

138     protected state: import("./TileState.js").default;
                                                  ~~~~~~~
```

The only thing we can do about them is to wait until the new version is out.

--- a/packages/map/package.json
+++ b/packages/map/package.json
@@ -38,7 +38,7 @@
     "@vaadin/component-base": "23.0.0-beta3",
     "@vaadin/vaadin-license-checker": "^2.1.0",
     "@vaadin/vaadin-themable-mixin": "23.0.0-beta3",
-    "ol": "6.11.0"
+    "ol": "6.12.1-dev.1645111187932"
   },
   "devDependencies": {
     "@esm-bundle/chai": "^4.3.4",

--- a/packages/map/test/typings/map.types.ts
+++ b/packages/map/test/typings/map.types.ts
@@ -1,0 +1,9 @@
+import '../../vaadin-map.js';
+import Map from 'ol/Map';
+
+const assertType = <TExpected>(actual: TExpected) => actual;
+
+const map = document.createElement('vaadin-map');
+
+// Should expose OL map instance
+assertType<Map>(map.configuration);

